nav2_dynamic_motion_model/include/nav2_dynamic_motion_model/motion_model_interface.hpp

```diff
@@ -0,0 +1,34 @@
+#ifndef NAV2_DYNAMIC_MOTION_MODEL__MOTION_MODEL_INTERFACE_HPP_
+#define NAV2_DYNAMIC_MOTION_MODEL__MOTION_MODEL_INTERFACE_HPP_
+
+#include <memory>
+#include <geometry_msgs/msg/pose.hpp>
+#include <geometry_msgs/msg/pose_array.hpp>
+#include <geometry_msgs/msg/twist.hpp>
+#include <nav2_dynamic_msgs/msg/obstacle.hpp>
+#include <nav2_dynamic_msgs/msg/obstacle_array.hpp>
+
+namespace nav2_dynamic_motion_model
+{
+
+class MotionModelInterface
+{
+public:
+    using Ptr = std::shared_ptr<MotionModelInterface>;
+    using ConstPtr = std::shared_ptr<const MotionModelInterface>;
+
+    virtual ~MotionModelInterface() = default;
+
+    virtual geometry_msgs::msg::Pose predictObstaclePose(
```

Comment:
Would we want a pose, or an obstacle message containing updated pose, velocity, propagated covariance information, etc?